<launch>
  <arg name="external_manager"    default="false"/>
  <arg name="manager"             default="realsense2_camera_manager"/>
  <arg name="serial_no"           default=""/>
  <arg name="usb_port_id"         default=""/>
  <arg name="device_type"         default=""/>
  <arg name="tf_prefix"           default=""/>
  <arg name="json_file_path"      default=""/>
  <arg name="rosbag_filename"     default=""/>
  <arg name="required"            default="false"/>

  <arg name="fisheye_width"       default="0"/>
  <arg name="fisheye_height"      default="0"/>
  <arg name="enable_fisheye"      default="true"/>
  <arg name="enable_fisheye1"     default="true"/>
  <arg name="enable_fisheye2"     default="true"/>

  <arg name="depth_width"         default="640"/>
  <arg name="depth_height"        default="480"/>
  <arg name="enable_depth"        default="true"/>

  <arg name="infra_width"         default="640"/>
  <arg name="infra_height"        default="480"/>
  <arg name="enable_infra1"       default="true"/>
  <arg name="enable_infra2"       default="true"/>

  <arg name="color_width"         default="640"/>
  <arg name="color_height"        default="480"/>
  <arg name="enable_color"        default="true"/>

  <arg name="fisheye_fps"         default="30"/>
  <arg name="depth_fps"           default="30"/>
  <arg name="infra_fps"           default="30"/>
  <arg name="color_fps"           default="30"/>
  <arg name="gyro_fps"            default="0"/>
  <arg name="accel_fps"           default="0"/>
  <arg name="enable_gyro"         default="true"/>
  <arg name="enable_accel"        default="true"/>

  <arg name="enable_pointcloud"   default="false"/>
  <arg name="pointcloud_texture_stream" default="RS2_STREAM_COLOR"/>  <!-- use RS2_STREAM_ANY to avoid using texture -->
  <arg name="pointcloud_texture_index"  default="0"/>

  <arg name="enable_sync"         default="false"/>
  <arg name="align_depth"         default="false"/>

  <arg name="base_frame_id"             default="$(arg tf_prefix)_link"/>
  <arg name="depth_frame_id"            default="$(arg tf_prefix)_depth_frame"/>
  <arg name="infra1_frame_id"           default="$(arg tf_prefix)_infra1_frame"/>
  <arg name="infra2_frame_id"           default="$(arg tf_prefix)_infra2_frame"/>
  <arg name="color_frame_id"            default="$(arg tf_prefix)_color_frame"/>
  <arg name="fisheye_frame_id"          default="$(arg tf_prefix)_fisheye_frame"/>
  <arg name="fisheye1_frame_id"         default="$(arg tf_prefix)_fisheye1_frame"/>
  <arg name="fisheye2_frame_id"         default="$(arg tf_prefix)_fisheye2_frame"/>
  <arg name="accel_frame_id"            default="$(arg tf_prefix)_accel_frame"/>
  <arg name="gyro_frame_id"             default="$(arg tf_prefix)_gyro_frame"/>
  <arg name="pose_frame_id"             default="$(arg tf_prefix)_pose_frame"/>

  <arg name="depth_optical_frame_id"    default="$(arg tf_prefix)_depth_optical_frame"/>
  <arg name="infra1_optical_frame_id"   default="$(arg tf_prefix)_infra1_optical_frame"/>
  <arg name="infra2_optical_frame_id"   default="$(arg tf_prefix)_infra2_optical_frame"/>
  <arg name="color_optical_frame_id"    default="$(arg tf_prefix)_color_optical_frame"/>
  <arg name="fisheye_optical_frame_id"  default="$(arg tf_prefix)_fisheye_optical_frame"/>
  <arg name="fisheye1_optical_frame_id" default="$(arg tf_prefix)_fisheye1_optical_frame"/>
  <arg name="fisheye2_optical_frame_id" default="$(arg tf_prefix)_fisheye2_optical_frame"/>
  <arg name="accel_optical_frame_id"    default="$(arg tf_prefix)_accel_optical_frame"/>
  <arg name="gyro_optical_frame_id"     default="$(arg tf_prefix)_gyro_optical_frame"/>
  <arg name="imu_optical_frame_id"      default="$(arg tf_prefix)_imu_optical_frame"/>
  <arg name="pose_optical_frame_id"     default="$(arg tf_prefix)_pose_optical_frame"/>

  <arg name="aligned_depth_to_color_frame_id"    default="$(arg tf_prefix)_aligned_depth_to_color_frame"/>
  <arg name="aligned_depth_to_infra1_frame_id"   default="$(arg tf_prefix)_aligned_depth_to_infra1_frame"/>
  <arg name="aligned_depth_to_infra2_frame_id"   default="$(arg tf_prefix)_aligned_depth_to_infra2_frame"/>
  <arg name="aligned_depth_to_fisheye_frame_id"  default="$(arg tf_prefix)_aligned_depth_to_fisheye_frame"/>
  <arg name="aligned_depth_to_fisheye1_frame_id" default="$(arg tf_prefix)_aligned_depth_to_fisheye1_frame"/>
  <arg name="aligned_depth_to_fisheye2_frame_id" default="$(arg tf_prefix)_aligned_depth_to_fisheye2_frame"/>

  <arg name="publish_tf"               default="true"/>
  <arg name="tf_publish_rate"          default="0"/> <!-- 0 - static transform -->

  <arg name="odom_frame_id"            default="$(arg tf_prefix)_odom_frame"/>
  <arg name="topic_odom_in"            default="$(arg tf_prefix)/odom_in"/>
  <arg name="calib_odom_file"          default=""/>
  <arg name="publish_odom_tf"          default="true"/>
  <arg name="filters"                  default=""/>
  <arg name="clip_distance"            default="-1"/>
  <arg name="linear_accel_cov"         default="0.01"/>
  <arg name="initial_reset"            default="false"/>
  <arg name="unite_imu_method"         default="none"/> <!-- Options are: [none, copy, linear_interpolation] -->
  <arg name="allow_no_texture_points"  default="false"/>


  <node unless="$(arg external_manager)" pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" output="screen" required="$(arg required)"/>
  <node pkg="nodelet" type="nodelet" name="realsense2_camera" args="load realsense2_camera/RealSenseNodeFactory $(arg manager)" required="$(arg required)">
    <param name="cam0/name"                type="str"  value="front"/>
    <param name="cam0/serial_no"           type="str"  value="909212111154"/>
    <param name="cam0/usb_port_id"         type="str"  value="$(arg usb_port_id)"/>
    <param name="cam0/device_type"         type="str"  value="t265"/>

    <param name="cam0/json_file_path"           type="str"  value="$(arg json_file_path)"/>
    <param name="cam0/rosbag_filename"          type="str"  value="$(arg rosbag_filename)"/>
    <param name="cam0/enable_pointcloud"        type="bool" value="$(arg enable_pointcloud)"/>
    <param name="cam0/pointcloud_texture_stream" type="str" value="$(arg pointcloud_texture_stream)"/>
    <param name="cam0/pointcloud_texture_index"  type="int" value="$(arg pointcloud_texture_index)"/>
    <param name="cam0/enable_sync"              type="bool" value="$(arg enable_sync)"/>
    <param name="cam0/align_depth"              type="bool" value="$(arg align_depth)"/>
    <param name="cam0/fisheye_width"            type="int"  value="$(arg fisheye_width)"/>
    <param name="cam0/fisheye_height"           type="int"  value="$(arg fisheye_height)"/>
    <param name="cam0/enable_fisheye"           type="bool" value="$(arg enable_fisheye)"/>
    <param name="cam0/enable_fisheye1"          type="bool" value="$(arg enable_fisheye1)"/>
    <param name="cam0/enable_fisheye2"          type="bool" value="$(arg enable_fisheye2)"/>
    <param name="cam0/depth_width"              type="int"  value="$(arg depth_width)"/>
    <param name="cam0/depth_height"             type="int"  value="$(arg depth_height)"/>
    <param name="cam0/enable_depth"             type="bool" value="$(arg enable_depth)"/>
    <param name="cam0/color_width"              type="int"  value="$(arg color_width)"/>
    <param name="cam0/color_height"             type="int"  value="$(arg color_height)"/>
    <param name="cam0/enable_color"             type="bool" value="$(arg enable_color)"/>
    <param name="cam0/infra_width"             type="int"  value="$(arg infra_width)"/>
    <param name="cam0/infra_height"            type="int"  value="$(arg infra_height)"/>
    <param name="cam0/enable_infra1"            type="bool" value="$(arg enable_infra1)"/>
    <param name="cam0/enable_infra2"            type="bool" value="$(arg enable_infra2)"/>
    <param name="cam0/fisheye_fps"              type="int"  value="$(arg fisheye_fps)"/>
    <param name="cam0/depth_fps"                type="int"  value="$(arg depth_fps)"/>
    <param name="cam0/infra_fps"               type="int"  value="$(arg infra_fps)"/>
    <param name="cam0/color_fps"                type="int"  value="$(arg color_fps)"/>
    <param name="cam0/gyro_fps"                 type="int"  value="$(arg gyro_fps)"/>
    <param name="cam0/accel_fps"                type="int"  value="$(arg accel_fps)"/>
    <param name="cam0/enable_gyro"              type="bool" value="$(arg enable_gyro)"/>
    <param name="cam0/enable_accel"             type="bool" value="$(arg enable_accel)"/>
    <param name="cam0/base_frame_id"            type="str"  value="$(arg base_frame_id)"/>
    <param name="cam0/depth_frame_id"           type="str"  value="$(arg depth_frame_id)"/>
    <param name="cam0/infra1_frame_id"          type="str"  value="$(arg infra1_frame_id)"/>
    <param name="cam0/infra2_frame_id"          type="str"  value="$(arg infra2_frame_id)"/>
    <param name="cam0/color_frame_id"           type="str"  value="$(arg color_frame_id)"/>
    <param name="cam0/fisheye_frame_id"         type="str"  value="$(arg fisheye_frame_id)"/>
    <param name="cam0/fisheye1_frame_id"        type="str"  value="$(arg fisheye1_frame_id)"/>
    <param name="cam0/fisheye2_frame_id"        type="str"  value="$(arg fisheye2_frame_id)"/>
    <param name="cam0/accel_frame_id"           type="str"  value="$(arg accel_frame_id)"/>
    <param name="cam0/gyro_frame_id"            type="str"  value="$(arg gyro_frame_id)"/>
    <param name="cam0/pose_frame_id"            type="str"  value="$(arg pose_frame_id)"/>
    <param name="cam0/depth_optical_frame_id"    type="str"  value="$(arg depth_optical_frame_id)"/>
    <param name="cam0/infra1_optical_frame_id"   type="str"  value="$(arg infra1_optical_frame_id)"/>
    <param name="cam0/infra2_optical_frame_id"   type="str"  value="$(arg infra2_optical_frame_id)"/>
    <param name="cam0/color_optical_frame_id"    type="str"  value="$(arg color_optical_frame_id)"/>
    <param name="cam0/fisheye_optical_frame_id"  type="str"  value="$(arg fisheye_optical_frame_id)"/>
    <param name="cam0/fisheye1_optical_frame_id" type="str"  value="$(arg fisheye1_optical_frame_id)"/>
    <param name="cam0/fisheye2_optical_frame_id" type="str"  value="$(arg fisheye2_optical_frame_id)"/>
    <param name="cam0/accel_optical_frame_id"    type="str"  value="$(arg accel_optical_frame_id)"/>
    <param name="cam0/gyro_optical_frame_id"     type="str"  value="$(arg gyro_optical_frame_id)"/>
    <param name="cam0/imu_optical_frame_id"      type="str"  value="$(arg imu_optical_frame_id)"/>
    <param name="cam0/pose_optical_frame_id"     type="str"  value="$(arg pose_optical_frame_id)"/>
    <param name="cam0/aligned_depth_to_color_frame_id"    type="str"  value="$(arg aligned_depth_to_color_frame_id)"/>
    <param name="cam0/aligned_depth_to_infra1_frame_id"   type="str"  value="$(arg aligned_depth_to_infra1_frame_id)"/>
    <param name="cam0/aligned_depth_to_infra2_frame_id"   type="str"  value="$(arg aligned_depth_to_infra2_frame_id)"/>
    <param name="cam0/aligned_depth_to_fisheye_frame_id"  type="str"  value="$(arg aligned_depth_to_fisheye_frame_id)"/>
    <param name="cam0/aligned_depth_to_fisheye1_frame_id" type="str"  value="$(arg aligned_depth_to_fisheye1_frame_id)"/>
    <param name="cam0/aligned_depth_to_fisheye2_frame_id" type="str"  value="$(arg aligned_depth_to_fisheye2_frame_id)"/>
    <param name="cam0/publish_tf"               type="bool"   value="$(arg publish_tf)"/>
    <param name="cam0/tf_publish_rate"          type="double" value="$(arg tf_publish_rate)"/>
    <param name="cam0/odom_frame_id"            type="str"  value="$(arg odom_frame_id)"/>
    <param name="cam0/topic_odom_in"            type="str"  value="$(arg topic_odom_in)"/>
    <param name="cam0/calib_odom_file"          type="str"    value="$(arg calib_odom_file)"/>
    <param name="cam0/publish_odom_tf"          type="bool" value="$(arg publish_odom_tf)"/>
    <param name="cam0/filters"                  type="str"    value="$(arg filters)"/>
    <param name="cam0/clip_distance"            type="double" value="$(arg clip_distance)"/>
    <param name="cam0/linear_accel_cov"         type="double" value="$(arg linear_accel_cov)"/>
    <param name="cam0/initial_reset"            type="bool"   value="$(arg initial_reset)"/>
    <param name="cam0/unite_imu_method"         type="str"    value="$(arg unite_imu_method)"/>
    <param name="cam0/allow_no_texture_points"  type="bool"   value="$(arg allow_no_texture_points)"/>

    <param name="cam1/name"                type="str"  value="right"/>
    <param name="cam1/serial_no"           type="str"  value="845412111005"/>
    <param name="cam1/usb_port_id"         type="str"  value="$(arg usb_port_id)"/>
    <param name="cam1/device_type"         type="str"  value="t265"/>

    <param name="cam1/json_file_path"           type="str"  value="$(arg json_file_path)"/>
    <param name="cam1/rosbag_filename"          type="str"  value="$(arg rosbag_filename)"/>
    <param name="cam1/enable_pointcloud"        type="bool" value="$(arg enable_pointcloud)"/>
    <param name="cam1/pointcloud_texture_stream" type="str" value="$(arg pointcloud_texture_stream)"/>
    <param name="cam1/pointcloud_texture_index"  type="int" value="$(arg pointcloud_texture_index)"/>
    <param name="cam1/enable_sync"              type="bool" value="$(arg enable_sync)"/>
    <param name="cam1/align_depth"              type="bool" value="$(arg align_depth)"/>
    <param name="cam1/fisheye_width"            type="int"  value="$(arg fisheye_width)"/>
    <param name="cam1/fisheye_height"           type="int"  value="$(arg fisheye_height)"/>
    <param name="cam1/enable_fisheye"           type="bool" value="$(arg enable_fisheye)"/>
    <param name="cam1/enable_fisheye1"          type="bool" value="$(arg enable_fisheye1)"/>
    <param name="cam1/enable_fisheye2"          type="bool" value="$(arg enable_fisheye2)"/>
    <param name="cam1/depth_width"              type="int"  value="$(arg depth_width)"/>
    <param name="cam1/depth_height"             type="int"  value="$(arg depth_height)"/>
    <param name="cam1/enable_depth"             type="bool" value="$(arg enable_depth)"/>
    <param name="cam1/color_width"              type="int"  value="$(arg color_width)"/>
    <param name="cam1/color_height"             type="int"  value="$(arg color_height)"/>
    <param name="cam1/enable_color"             type="bool" value="$(arg enable_color)"/>
    <param name="cam1/infra_width"             type="int"  value="$(arg infra_width)"/>
    <param name="cam1/infra_height"            type="int"  value="$(arg infra_height)"/>
    <param name="cam1/enable_infra1"            type="bool" value="$(arg enable_infra1)"/>
    <param name="cam1/enable_infra2"            type="bool" value="$(arg enable_infra2)"/>
    <param name="cam1/fisheye_fps"              type="int"  value="$(arg fisheye_fps)"/>
    <param name="cam1/depth_fps"                type="int"  value="$(arg depth_fps)"/>
    <param name="cam1/infra_fps"               type="int"  value="$(arg infra_fps)"/>
    <param name="cam1/color_fps"                type="int"  value="$(arg color_fps)"/>
    <param name="cam1/gyro_fps"                 type="int"  value="$(arg gyro_fps)"/>
    <param name="cam1/accel_fps"                type="int"  value="$(arg accel_fps)"/>
    <param name="cam1/enable_gyro"              type="bool" value="$(arg enable_gyro)"/>
    <param name="cam1/enable_accel"             type="bool" value="$(arg enable_accel)"/>
    <param name="cam1/base_frame_id"            type="str"  value="$(arg base_frame_id)"/>
    <param name="cam1/depth_frame_id"           type="str"  value="$(arg depth_frame_id)"/>
    <param name="cam1/infra1_frame_id"          type="str"  value="$(arg infra1_frame_id)"/>
    <param name="cam1/infra2_frame_id"          type="str"  value="$(arg infra2_frame_id)"/>
    <param name="cam1/color_frame_id"           type="str"  value="$(arg color_frame_id)"/>
    <param name="cam1/fisheye_frame_id"         type="str"  value="$(arg fisheye_frame_id)"/>
    <param name="cam1/fisheye1_frame_id"        type="str"  value="$(arg fisheye1_frame_id)"/>
    <param name="cam1/fisheye2_frame_id"        type="str"  value="$(arg fisheye2_frame_id)"/>
    <param name="cam1/accel_frame_id"           type="str"  value="$(arg accel_frame_id)"/>
    <param name="cam1/gyro_frame_id"            type="str"  value="$(arg gyro_frame_id)"/>
    <param name="cam1/pose_frame_id"            type="str"  value="$(arg pose_frame_id)"/>
    <param name="cam1/depth_optical_frame_id"    type="str"  value="$(arg depth_optical_frame_id)"/>
    <param name="cam1/infra1_optical_frame_id"   type="str"  value="$(arg infra1_optical_frame_id)"/>
    <param name="cam1/infra2_optical_frame_id"   type="str"  value="$(arg infra2_optical_frame_id)"/>
    <param name="cam1/color_optical_frame_id"    type="str"  value="$(arg color_optical_frame_id)"/>
    <param name="cam1/fisheye_optical_frame_id"  type="str"  value="$(arg fisheye_optical_frame_id)"/>
    <param name="cam1/fisheye1_optical_frame_id" type="str"  value="$(arg fisheye1_optical_frame_id)"/>
    <param name="cam1/fisheye2_optical_frame_id" type="str"  value="$(arg fisheye2_optical_frame_id)"/>
    <param name="cam1/accel_optical_frame_id"    type="str"  value="$(arg accel_optical_frame_id)"/>
    <param name="cam1/gyro_optical_frame_id"     type="str"  value="$(arg gyro_optical_frame_id)"/>
    <param name="cam1/imu_optical_frame_id"      type="str"  value="$(arg imu_optical_frame_id)"/>
    <param name="cam1/pose_optical_frame_id"     type="str"  value="$(arg pose_optical_frame_id)"/>
    <param name="cam1/aligned_depth_to_color_frame_id"    type="str"  value="$(arg aligned_depth_to_color_frame_id)"/>
    <param name="cam1/aligned_depth_to_infra1_frame_id"   type="str"  value="$(arg aligned_depth_to_infra1_frame_id)"/>
    <param name="cam1/aligned_depth_to_infra2_frame_id"   type="str"  value="$(arg aligned_depth_to_infra2_frame_id)"/>
    <param name="cam1/aligned_depth_to_fisheye_frame_id"  type="str"  value="$(arg aligned_depth_to_fisheye_frame_id)"/>
    <param name="cam1/aligned_depth_to_fisheye1_frame_id" type="str"  value="$(arg aligned_depth_to_fisheye1_frame_id)"/>
    <param name="cam1/aligned_depth_to_fisheye2_frame_id" type="str"  value="$(arg aligned_depth_to_fisheye2_frame_id)"/>
    <param name="cam1/publish_tf"               type="bool"   value="$(arg publish_tf)"/>
    <param name="cam1/tf_publish_rate"          type="double" value="$(arg tf_publish_rate)"/>
    <param name="cam1/odom_frame_id"            type="str"  value="$(arg odom_frame_id)"/>
    <param name="cam1/topic_odom_in"            type="str"  value="$(arg topic_odom_in)"/>
    <param name="cam1/calib_odom_file"          type="str"    value="$(arg calib_odom_file)"/>
    <param name="cam1/publish_odom_tf"          type="bool" value="$(arg publish_odom_tf)"/>
    <param name="cam1/filters"                  type="str"    value="$(arg filters)"/>
    <param name="cam1/clip_distance"            type="double" value="$(arg clip_distance)"/>
    <param name="cam1/linear_accel_cov"         type="double" value="$(arg linear_accel_cov)"/>
    <param name="cam1/initial_reset"            type="bool"   value="$(arg initial_reset)"/>
    <param name="cam1/unite_imu_method"         type="str"    value="$(arg unite_imu_method)"/>
    <param name="cam1/allow_no_texture_points"  type="bool"   value="$(arg allow_no_texture_points)"/>

    <param name="cam2/name"                type="str"  value="left"/>
    <param name="cam2/serial_no"           type="str"  value="908412111064"/>
    <param name="cam2/usb_port_id"         type="str"  value="$(arg usb_port_id)"/>
    <param name="cam2/device_type"         type="str"  value="t265"/>

    <param name="cam2/json_file_path"           type="str"  value="$(arg json_file_path)"/>
    <param name="cam2/rosbag_filename"          type="str"  value="$(arg rosbag_filename)"/>
    <param name="cam2/enable_pointcloud"        type="bool" value="$(arg enable_pointcloud)"/>
    <param name="cam2/pointcloud_texture_stream" type="str" value="$(arg pointcloud_texture_stream)"/>
    <param name="cam2/pointcloud_texture_index"  type="int" value="$(arg pointcloud_texture_index)"/>
    <param name="cam2/enable_sync"              type="bool" value="$(arg enable_sync)"/>
    <param name="cam2/align_depth"              type="bool" value="$(arg align_depth)"/>
    <param name="cam2/fisheye_width"            type="int"  value="$(arg fisheye_width)"/>
    <param name="cam2/fisheye_height"           type="int"  value="$(arg fisheye_height)"/>
    <param name="cam2/enable_fisheye"           type="bool" value="$(arg enable_fisheye)"/>
    <param name="cam2/enable_fisheye1"          type="bool" value="$(arg enable_fisheye1)"/>
    <param name="cam2/enable_fisheye2"          type="bool" value="$(arg enable_fisheye2)"/>
    <param name="cam2/depth_width"              type="int"  value="$(arg depth_width)"/>
    <param name="cam2/depth_height"             type="int"  value="$(arg depth_height)"/>
    <param name="cam2/enable_depth"             type="bool" value="$(arg enable_depth)"/>
    <param name="cam2/color_width"              type="int"  value="$(arg color_width)"/>
    <param name="cam2/color_height"             type="int"  value="$(arg color_height)"/>
    <param name="cam2/enable_color"             type="bool" value="$(arg enable_color)"/>
    <param name="cam2/infra_width"             type="int"  value="$(arg infra_width)"/>
    <param name="cam2/infra_height"            type="int"  value="$(arg infra_height)"/>
    <param name="cam2/enable_infra1"            type="bool" value="$(arg enable_infra1)"/>
    <param name="cam2/enable_infra2"            type="bool" value="$(arg enable_infra2)"/>
    <param name="cam2/fisheye_fps"              type="int"  value="$(arg fisheye_fps)"/>
    <param name="cam2/depth_fps"                type="int"  value="$(arg depth_fps)"/>
    <param name="cam2/infra_fps"               type="int"  value="$(arg infra_fps)"/>
    <param name="cam2/color_fps"                type="int"  value="$(arg color_fps)"/>
    <param name="cam2/gyro_fps"                 type="int"  value="$(arg gyro_fps)"/>
    <param name="cam2/accel_fps"                type="int"  value="$(arg accel_fps)"/>
    <param name="cam2/enable_gyro"              type="bool" value="$(arg enable_gyro)"/>
    <param name="cam2/enable_accel"             type="bool" value="$(arg enable_accel)"/>
    <param name="cam2/base_frame_id"            type="str"  value="$(arg base_frame_id)"/>
    <param name="cam2/depth_frame_id"           type="str"  value="$(arg depth_frame_id)"/>
    <param name="cam2/infra1_frame_id"          type="str"  value="$(arg infra1_frame_id)"/>
    <param name="cam2/infra2_frame_id"          type="str"  value="$(arg infra2_frame_id)"/>
    <param name="cam2/color_frame_id"           type="str"  value="$(arg color_frame_id)"/>
    <param name="cam2/fisheye_frame_id"         type="str"  value="$(arg fisheye_frame_id)"/>
    <param name="cam2/fisheye1_frame_id"        type="str"  value="$(arg fisheye1_frame_id)"/>
    <param name="cam2/fisheye2_frame_id"        type="str"  value="$(arg fisheye2_frame_id)"/>
    <param name="cam2/accel_frame_id"           type="str"  value="$(arg accel_frame_id)"/>
    <param name="cam2/gyro_frame_id"            type="str"  value="$(arg gyro_frame_id)"/>
    <param name="cam2/pose_frame_id"            type="str"  value="$(arg pose_frame_id)"/>
    <param name="cam2/depth_optical_frame_id"    type="str"  value="$(arg depth_optical_frame_id)"/>
    <param name="cam2/infra1_optical_frame_id"   type="str"  value="$(arg infra1_optical_frame_id)"/>
    <param name="cam2/infra2_optical_frame_id"   type="str"  value="$(arg infra2_optical_frame_id)"/>
    <param name="cam2/color_optical_frame_id"    type="str"  value="$(arg color_optical_frame_id)"/>
    <param name="cam2/fisheye_optical_frame_id"  type="str"  value="$(arg fisheye_optical_frame_id)"/>
    <param name="cam2/fisheye1_optical_frame_id" type="str"  value="$(arg fisheye1_optical_frame_id)"/>
    <param name="cam2/fisheye2_optical_frame_id" type="str"  value="$(arg fisheye2_optical_frame_id)"/>
    <param name="cam2/accel_optical_frame_id"    type="str"  value="$(arg accel_optical_frame_id)"/>
    <param name="cam2/gyro_optical_frame_id"     type="str"  value="$(arg gyro_optical_frame_id)"/>
    <param name="cam2/imu_optical_frame_id"      type="str"  value="$(arg imu_optical_frame_id)"/>
    <param name="cam2/pose_optical_frame_id"     type="str"  value="$(arg pose_optical_frame_id)"/>
    <param name="cam2/aligned_depth_to_color_frame_id"    type="str"  value="$(arg aligned_depth_to_color_frame_id)"/>
    <param name="cam2/aligned_depth_to_infra1_frame_id"   type="str"  value="$(arg aligned_depth_to_infra1_frame_id)"/>
    <param name="cam2/aligned_depth_to_infra2_frame_id"   type="str"  value="$(arg aligned_depth_to_infra2_frame_id)"/>
    <param name="cam2/aligned_depth_to_fisheye_frame_id"  type="str"  value="$(arg aligned_depth_to_fisheye_frame_id)"/>
    <param name="cam2/aligned_depth_to_fisheye1_frame_id" type="str"  value="$(arg aligned_depth_to_fisheye1_frame_id)"/>
    <param name="cam2/aligned_depth_to_fisheye2_frame_id" type="str"  value="$(arg aligned_depth_to_fisheye2_frame_id)"/>
    <param name="cam2/publish_tf"               type="bool"   value="$(arg publish_tf)"/>
    <param name="cam2/tf_publish_rate"          type="double" value="$(arg tf_publish_rate)"/>
    <param name="cam2/odom_frame_id"            type="str"  value="$(arg odom_frame_id)"/>
    <param name="cam2/topic_odom_in"            type="str"  value="$(arg topic_odom_in)"/>
    <param name="cam2/calib_odom_file"          type="str"    value="$(arg calib_odom_file)"/>
    <param name="cam2/publish_odom_tf"          type="bool" value="$(arg publish_odom_tf)"/>
    <param name="cam2/filters"                  type="str"    value="$(arg filters)"/>
    <param name="cam2/clip_distance"            type="double" value="$(arg clip_distance)"/>
    <param name="cam2/linear_accel_cov"         type="double" value="$(arg linear_accel_cov)"/>
    <param name="cam2/initial_reset"            type="bool"   value="$(arg initial_reset)"/>
    <param name="cam2/unite_imu_method"         type="str"    value="$(arg unite_imu_method)"/>
    <param name="cam2/allow_no_texture_points"  type="bool"   value="$(arg allow_no_texture_points)"/>
 
  </node>
</launch>
